core/flyout_base.js

```diff
@@ -801,12 +801,17 @@ Blockly.Flyout.prototype.createBlock = function(originalBlock) {
 
   if (Blockly.Events.isEnabled()) {
     Blockly.Events.setGroup(true);
-    Blockly.Events.fire(new Blockly.Events.Create(newBlock));
     // Fire a VarCreate event for each (if any) new variable created.
     for (var i = 0; i < newVariables.length; i++) {
       var thisVariable = newVariables[i];
       Blockly.Events.fire(new Blockly.Events.VarCreate(thisVariable));
     }
+
+    // Block events come after var events, in case they refer to newly created
+    // variables.
+    Blockly.Events.fire(new Blockly.Events.Create(newBlock));
+
+    // Why don't we need to setgroup to false??
```

Comment:
Creating a block this way always involves a drag, and this should be called from Gesture or Block Dragger. (You'll want to check me on that.)

The drag continues, and includes moving the block, which should also be grouped with the create events. You'll probably be able to find the matching set group in the code that ends a drag.

Please verify that the group is no longer set at the end of the drag. If you find anything that seems fishy when looking into it, please file a bug to track.